R/geom_recessions.R

```diff
@@ -1,352 +1,345 @@
+#'  rectangle and text geoms, respectively.
+#'
+#'@section Important notes: If \code{show.legend = TRUE} you must place any
+#'  categorical aesthetics (e.g. color, size) specific to the primary data in
+#'  the geom(s) used to display that data. Otherwise, the legend will inherit
+#'  aesthetics from geom_recessions.
+#'
+#'  It is best to place this object before your primary geom (likely
+#'  \code{geom_line()}) in your code, so that ggplot draws it behind the primary
+#'  data being drawn.
+#'
+#'@section Default color: The CMAP color palette gray used for recessions is
+#'  \code{#e3e8eb}. The rectangle geom has default fill and alpha values of
+#'  \code{#002d49} and \code{0.11} built into the function. These replicate the
+#'  palette color at the highest possible transparency. This is done because
+#'  there is no known way to place the recession geom behind the graph's grid
+#'  lines. The default therefore produces the approved CMAP color while altering
+#'  the appearance of any overlapping grid lines as little as possible. These
+#'  can be overridden, but separately. Override fill using the top-level
+#'  argument, as in \code{fill = "red"}. Override alpha within rect_aes as in
+#'  \code{rect_aes = list(alpha = 0.5)}. Color and alpha were calculated using
+#'  the hints found here:
+#'  \url{https://stackoverflow.com/questions/6672374/convert-rgb-to-rgba-over-white}.
+#'
+#'
+#'@section Under the hood: This function calls two custom geoms, constructed
+#'  with ggproto. The custom GeomRecessions and GeomRecessionsText are modified
+#'  versions of GeomRect and GeomText, respectively. The only variations to each
+#'  occur in \code{default_aes}, \code{required_aes}, and \code{setup_data}
+#'  arguments. These variations allow the the primary dataframe (specified in
+#'  \code{ggplot(data = XXX)}) to filter the recessions displayed.
+#'
+#'
+#' @examples
+#' grp_goods <- dplyr::filter(grp_over_time, category == "Goods-Producing")
+#' grp_goods <- dplyr::mutate(grp_goods, year2 = as.Date(lubridate::date_decimal(year)))
+#'
+#' # INTEGER X AXIS:
+#' ggplot(grp_over_time, aes(x = year, y = realgrp, color = cluster)) +
+#'   geom_recessions() +
+#'   geom_line() +
+#'   scale_x_continuous("Year") +
+#'   theme_minimal()
+#'
+#' # DATE X AXIS:
+#' ggplot(data = grp_goods,
+#' mapping = aes(x = year2, y = realgrp, color = cluster)) +
+#'   geom_recessions(xformat = "date", ) +
+#'   geom_line() +
+#'   scale_x_date("Year") +
+#'   theme_minimal()
+#'
+#' # MODIFIED AESTHETICS:
+#' ggplot(grp_over_time, aes(x = year, y = realgrp)) +
+#'   geom_recessions(show.legend = TRUE, fill = "blue", text = FALSE,
+#'                   rect_aes = list(alpha = 1, color = "red")) +
+#'   geom_line(aes(color = cluster)) +
+#'   scale_x_continuous("Year") +
+#'   theme_minimal()
+#'
+#'
+#' # BELOW EXAMPLES SHOW MORE THAN 1 RECESSION
+#' df <- data.frame(date_dec=1800:2020, value=rnorm(442), var=c(rep("A", 221), rep("B", 221)))

```

Comment:
Let's simplify this to a more even number of years! The examples use a 4 year spread. If we change this to 1950 to 1999, we can remove the filter lines below.

Reply:
I fixed.